core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -418,7 +418,7 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    * yielding the value of the first effect to succeed with a value.
    * Losers of the race will be interrupted immediately
    */
-  def firstSuccessOf[R1 <: R, E1 >: E, A1 >: A](rest: Iterable[ZIO[R1, E1, A1]]): ZIO[R1, E1, A1] =
+  final def firstSuccessOf[R1 <: R, E1 >: E, A1 >: A](rest: Iterable[ZIO[R1, E1, A1]]): ZIO[R1, E1, A1] =
```

Comment:
Same here, let's focus on by-name parameters only, as it is already quite controversial.

Reply:
Yeah, makes sense @regiskuckaertz. This will need be properly split into several PRs focusing and addressing different issues